src/routes/goals/handlers.js

```diff
@@ -227,3 +228,40 @@ export async function retrieveGoalByIdAndRecipient(req, res) {
     await handleErrors(req, res, error, `${logContext}:RETRIEVE_GOAL_BY_ID_AND_RECIPIENT`);
   }
 }
+
+export async function getSimilarGoalsForRecipient(req, res) {
+  const recipientId = parseInt(req.params.recipient_id, 10);
+
+  if (Number.isNaN(recipientId)) {
+    return res.status(400).send('Recipient ID must be an integer');
```

Comment:
one small thing: we have middleware for checking the params in this manner - you'd still have to parseInt here but you wouldn't need the if statement or a test for it... I know we're pretty inconsistent about using the middleware

Reply:
🙏 d25bd1971207b3c867a1bba6c8f1552ab92d4b2b

Reply:
er, fc66a49b7d89377db46c1c268725252eb0b3ec91

Reply:
Usage might be a little off: 

We use the middleware in routes/goals/index.js like

```js
router.get(
  '/similar/:recipient_id', 
  (req, res, next) => checkIdParam(req, res, next, 'recipient_id'),
  transactionWrapper(getSimilarGoalsForRecipient)
);
```

then, in getSimilarGoalsForRecipient you'd just do one of these:

```js
const recipientId = parseInt(req.params.recipient_id, DECIMAL_BASE); 
```

Reply:
091bf508c2647b77908e3398e6efce1e9d48ef9b

oh jeez, I feel dumb. thank you